pmlc/dialect/pxa/transforms/stencil.cc

```diff
@@ -253,8 +302,209 @@ void Stencil::CollectStrideOneIndices() {
   }
 }
 
+// Test if idx in tensors[tensor_idx] is stride one index
+bool Stencil::IsStrideOne(mlir::BlockArgument idx, unsigned tensor_idx) {
+  return strideOne[tensor_idx].find(idx) != strideOne[tensor_idx].end();
+}
+
+// Test if idx in the tensors are stride one
+bool Stencil::ValidateStrideOne(mlir::BlockArgument idx, unsigned matrix_idx) {
+  switch (matrix_idx) {
+  case 0:
+    // Test if M is stride one for B(3) and C(2)
+    return IsStrideOne(idx, tensorsOrder[1]) &&
+           IsStrideOne(idx, tensorsOrder[2]);
+  case 1:
+    // N is not restricted for stride one
+    return true;
+  case 2:
+    // Test if K is stride one for A(0)
+    return IsStrideOne(idx, tensorsOrder[0]);
+  default:
+    assert(!"Wrong matrix_idx");
```

Comment:
This should probably be llvm_unreachable along with the other assert.

Reply:
Fixed.